### ISSUE TITLE
Errors after migration

### ISSUE THREAD

**Issue by vovpff** [YetiForceCompany/YetiForceCRM#1029]
Last week I established fresh apache 2.4 with PHP 5.6.9 an MySQL server and installed Yetiforce system. I like it too much!

For this time I have vtiger 6.2.0. This system works well but I need features what has YetiForce. Yestrday I made migration from Vtiger 6.2.0 to Yetiforce 2.0.0. Migration was succesfull, but after login window I has only this error: {"success":false,"error":{"code":"result is not an object","message":"result is not an object"}}

Any ideas how to solve it?

**Comment by bpabiszczak** [comment]
If you can send an email directly to me with access data to a testing version for your server [a link to CRM, login\password to CRM, access data to FTP] because we will be able to check what's the reason. Do not paste your access data on GitHub! My email: help@yetiforce.com

**Comment by vovpff** [comment]
Sorry bpabiszczak. My system established only in company network and there is no acess from internet. What more detailed information I can prepare for you?

**Comment by bpabiszczak** [comment]
1. First it needs to be specified whether this problem is only related to your server or to the system after migration. It’s best to install the system on WAMP or somewhere on an external hosting and verify it.

2. If the problem is on a server side, you need to contact your administrator but if it also occurs on a different server, then you have the following options:

• Search for the solution by enabling logs: https://yetiforce.com/en/ideas,-changes,-bugs/bugs/115-debugging.html

• Send us a copy of your system and in a spare moment we will verify what is the reason. We will run a copy [base+files can be send to the following address: help@yetiforce.com], on a virtual system with a clean LAMP configuration and search for the reason.

**Comment by vovpff** [comment]
My logs:

SQL debug:

<hr>
(mysql): SELECT * FROM `vtiger_bruteforce` LIMIT 1 &nbsp; 
<hr>
<hr>
(mysql): SELECT count(login_id) as cn FROM `vtiger_loginhistory` vlh WHERE STATUS = 'Failed login' &amp;&amp; user_ip = '192.168.1.114' &amp;&amp; unblock = 0 &amp;&amp; (UNIX_TIMESTAMP(vlh.login_time) - UNIX_TIMESTAMP(ADDDATE('2015-06-08 12:14:25', INTERVAL -15 MINUTE))) &gt; 0 &nbsp; 
<hr>
1054: Unknown column 'unblock' in 'where clause'<br>
<pre align=left> &nbsp;  &nbsp;  &nbsp;  &nbsp; <font face="Courier New,Courier">ADOConnection._Execute(SELECT count(login_id) as cn FROM `vtiger_loginhistory` vlh WHERE STATUS = 'Failed login' &amp;&amp; user_ip = '192.168.1.114' &amp;&amp; unbloc...)</font><font color=#808080 size=-1> % line 1029, file: <a href="file:/D:\server\Apache\htdocs\yt\libraries\adodb\adodb.inc.php">adodb.inc.php</a></font>
 &nbsp;  &nbsp;  &nbsp; <font face="Courier New,Courier">ADOConnection.Execute(SELECT count(login_id) as cn FROM `vtiger_loginhistory` vlh WHERE STATUS = 'Failed login' &amp;&amp; user_ip = ? &amp;&amp; unblock = 0 &amp;&amp; (UNIX..., Array[2])</font><font color=#808080 size=-1> % line  333, file: <a href="file:/D:\server\Apache\htdocs\yt\include\database\PearDatabase.php">PearDatabase.php</a></font>
 &nbsp;  &nbsp; <font face="Courier New,Courier">PearDatabase.pquery(SELECT count(login_id) as cn FROM `vtiger_loginhistory` vlh WHERE STATUS = 'Failed login' &amp;&amp; user_ip = ? &amp;&amp; unblock = 0 &amp;&amp; (UNIX..., Array[2])</font><font color=#808080 size=-1> % line   87, file: <a href="file:/D:\server\Apache\htdocs\yt\modules\Settings\BruteForce\models\Module.php">Module.php</a></font>
 &nbsp; <font face="Courier New,Courier">Settings_BruteForce_Module_Model.checkBlocked()</font><font color=#808080 size=-1> % line   31, file: <a href="file:/D:\server\Apache\htdocs\yt\modules\Users\actions\Login.php">Login.php</a></font>
<font face="Courier New,Courier">Users_Login_Action.process(Object:Vtiger_Request)</font><font color=#808080 size=-1> % line  194, file: <a href="file:/D:\server\Apache\htdocs\yt\include\main\WebUI.php">WebUI.php</a></font>
</pre>{"success":false,"error":{"code":"result is not an object","message":"result is not an object"}}

system.log:

06/08/15 12:11:43,686 [7904] INFO VT - PearDatabase ->ADODB error  Query Failed:SELECT * FROM `vtiger_bruteforce` LIMIT 1::->[1146]Table 'vtigercrm540.vtiger_bruteforce' doesn't exist
06/08/15 12:14:14,140 [7904] INFO VT - PearDatabase ->ADODB disconnect
06/08/15 12:14:14,341 [7904] DEBUG VT - Prepared sql query being executed : select * from vtiger_organizationdetails
06/08/15 12:14:16,191 [7904] INFO VT - PearDatabase ->ADODB disconnect
06/08/15 12:14:16,381 [7904] DEBUG VT - Prepared sql query being executed : select * from vtiger_organizationdetails
06/08/15 12:14:25,135 [7904] INFO VT - PearDatabase ->ADODB disconnect
06/08/15 12:14:25,365 [7904] DEBUG VT - Prepared sql query being executed : SELECT * FROM `vtiger_bruteforce` LIMIT 1
06/08/15 12:14:25,365 [7904] DEBUG VT - Prepared sql query being executed : SELECT count(login_id) as cn FROM `vtiger_loginhistory` vlh WHERE STATUS = 'Failed login' && user_ip = ? && unblock = 0 && (UNIX_TIMESTAMP(vlh.login_time) - UNIX_TIMESTAMP(ADDDATE(?, INTERVAL -15 MINUTE))) > 0
06/08/15 12:14:25,365 [7904] DEBUG VT - Prepared sql query parameters : [192.168.1.114,2015-06-08 12:14:25]
06/08/15 12:14:25,375 [7904] INFO VT - PearDatabase ->ADODB error  Query Failed:SELECT count(login_id) as cn FROM `vtiger_loginhistory` vlh WHERE STATUS = 'Failed login' && user_ip = ? && unblock = 0 && (UNIX_TIMESTAMP(vlh.login_time) - UNIX_TIMESTAMP(ADDDATE(?, INTERVAL -15 MINUTE))) > 0::->[1054]Unknown column 'unblock' in 'where clause'

Looks like after migration there is no some tables and data in database.

**Comment by vovpff** [comment]
For this time I made next steps.

I executed migration VT620_to_YT***.sql files from folder "migrate_schema' in next order: delete -> create -> update manually. After that I has logged in to system but some modules not responding. Looks like migration module not woking corect.

**Comment by waran70** [comment]
Here is the same. Migration dont work. Got two friends who tried migrate 6.2  - it not works.
Updated, create, delete sql's needs to be revised (especially update and create). I posted to email list following:
migration script VT602_to_YT_create.sql contains following script:

CREATE TABLE IF NOT EXISTS `vtiger_calculationsproductrel` (
  `id` int(19) DEFAULT NULL,
  `productid` int(19) DEFAULT NULL,
  `sequence_no` int(4) DEFAULT NULL,
  `quantity` decimal(25,3) DEFAULT NULL,
  `listprice` decimal(27,8) DEFAULT NULL,
  `comment` text,
  `description` text,
  `lineitem_id` int(11) NOT NULL AUTO_INCREMENT,
  `incrementondel` int(19) DEFAULT NULL,
  `rbh` decimal(10,2) DEFAULT NULL,
  `purchase` decimal(10,2) DEFAULT NULL,
  `margin` decimal(10,2) DEFAULT NULL,
  `marginp` decimal(10,2) DEFAULT NULL,
  PRIMARY KEY (`lineitem_id`),
  KEY `calculationsproductrel_id_idx` (`id`),
  KEY `calculationsproductrel_productid_idx` (`productid`),
  CONSTRAINT `vtiger_calculationsproductrel_ibfk_1` FOREIGN KEY (`id`) REFERENCES `vtiger_calculations` (`calculationsid`) ON DELETE CASCADE
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

But it don't have any script creating table vtiger_calculations !
so the migraion HAVE to fail do to unable to create constraint and foreign key.

It will be worth to look at migration scripts again.

**Comment by vovpff** [comment]
I executed mirgation one more time with different action sequences.

1. Importing sql file with database of my vtiger 6.2 directly by HeidiSQL
2. Executing VT620_to_YT_delete.sql
3. Executing VT620_to_YT_create.sql
4. Executing VT620_to_YT_update.sql
5. Migration

Migration was completed succesfull with message:

array (
  0 => 
  array (
    'file' => 'D:\\server\\Apache\\htdocs\\yt\\include\\database\\PearDatabase.php',
    'line' => 277,
    'function' => 'checkError',
  ),
  1 => 
  array (
    'file' => 'D:\\server\\Apache\\htdocs\\yt\\install\\migrate_schema\\VT620_to_YT.php',
    'line' => 4252,
    'function' => 'query',
  ),
  2 => 
  array (
    'file' => 'D:\\server\\Apache\\htdocs\\yt\\install\\migrate_schema\\VT620_to_YT.php',
    'line' => 4083,
    'function' => 'DeleteParenttabrel',
  ),
  3 => 
  array (
    'file' => 'D:\\server\\Apache\\htdocs\\yt\\install\\migrate_schema\\VT620_to_YT.php',
    'line' => 3908,
    'function' => 'DeleteAll',
  ),
  4 => 
  array (
    'file' => 'D:\\server\\Apache\\htdocs\\yt\\install\\migrate_schema\\VT620_to_YT.php',
    'line' => 105,
    'function' => 'removeModules',
  ),
  5 => 
  array (
    'file' => 'D:\\server\\Apache\\htdocs\\yt\\install\\models\\InitSchema.php',
    'line' => 162,
    'function' => 'process',
  ),
  6 => 
  array (
    'file' => 'D:\\server\\Apache\\htdocs\\yt\\install\\views\\Index.php',
    'line' => 348,
    'function' => 'executeMigrationSchema',
  ),
  7 => 
  array (
    'file' => 'D:\\server\\Apache\\htdocs\\yt\\install\\views\\Index.php',
    'line' => 89,
    'function' => 'mStep3',
  ),
  8 => 
  array (
    'file' => 'D:\\server\\Apache\\htdocs\\yt\\install\\Install.php',
    'line' => 18,
    'function' => 'process',
  ),
)

After this migration i don't have problems with login but everytime has error message "Handler not found" in different modules.

And there is same situation like in issue ttps://github.com/YetiForceCompany/YetiForceCRM/issues/1032 but add widget button is active at "My home page". This buttons give me ability to add widgets what was used by me in vtiger but clicking for this lines do not made any difference.

**Comment by waran70** [comment]
It looks like migration is prepared on fresh untouched installation of vtiger 6.2.
This seems to be far away from expectations and logic as who is installing CRM such as vtiger and not even adding custom field? 
Again, it looks like update/create sql has some issues. Do try on various databases before releasing migration scripts (until this is kind of a way to get new customers on board O.o)

**Comment by bpabiszczak** [comment]
1. You are mixing many problems and many error messages in one issue and to be honest I’m lost. I’d close it because I don’t understand what you write [and it’s not because of the language barrier].

2. If we didn’t add a few tables during migration, everyone would have this problem. However, we haven’t noticed it so either you describe it step-by-step or we won’t be able to help you. Not having a table might have different reasons. It may also result from a situation that we didn’t foresee and we will endlessly search for a solution if you don’t show us the way.

3. In order to eliminate a problem with a server, we can create a clean version of Vtiger 6.2 for migration and that’s how we will be able to see what might be a problem.

4. Clean Vtiger doesn’t mean that there are no fields added, but it means that no system files have been modified. If you want, you can add fields, there is no difference to me.

5. When you write that you didn’t manage to do or complete something isn’t helpful to me. I need logs, a step-by step scenario but the most eagerly we accept a direct fix of a migration script.

6. Migration scripts don’t have to contain creation of a table to create it because the migration script loads XML files of a module and there are mechanisms creating SQL structure for modules.

7. I described it in a separate thread. At the beginning of a script you need to disable key checking because if an error during a creation of table A by a table B appears then you will know that there is a problem with the configuration. Waran70 please add more substantive comments because your descriptions are more harmful than beneficial.

8. We don’t take any responsibility if you perform migration in a different way. You do it at your own risk.

9. And I ask you again, don’t mix problems. The problem with a table is different to the problem with a migration and the problem with widgets is also a separate issue.

10. Waran70 your last comment doesn’t add anything to this discussion so I will ignore it.
